es-app/src/main.cpp

```diff
@@ -355,8 +355,8 @@ int main(int argc, char* argv[])
 		int deltaTime = curTime - lastTime;
 		lastTime = curTime;
 
-		// cap deltaTime
-		if((deltaTime > PowerSaver::getTimeout() && PowerSaver::getTimeout() > 0) || deltaTime < 0)
+		// cap deltaTime if it goes over non-zero Screensaver timeout or if negative
+		if((deltaTime > PowerSaver::getTimeout() + 100 && PowerSaver::getTimeout() > 0) || deltaTime < 0)
```

Comment:
Also, this first part should only be executed if we're not in standby (which may defeat the purpose). But otherwise all other components' internal clock won't be updated as expected, and issues will arise - namely the videos in the screensaver won't switch, at least.